### PR TITLE
Fix hold note approach fade out

### LINKED ISSUE #269: Fix hold note approach fade out
As an editor, I want the approach for hold notes to behave the same as normal note so that the two are consistent.

For normal note, the approach fades with the note after hit time, but for hold note, approach currently does not fade with the release time, alpha stays 1 for the approach and then disappears.

Normal note fading:
image

Hold note fading:
image

### PR DESCRIPTION
**Pull request by Xenocidel**
Closes #269

### CODE REVIEW

File: S2VX.Game.Tests/VisualTests/EditorHoldNoteTests.cs [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+        private FramedClock FramedClock { get; set; }
+
+        [BackgroundDependencyLoader]
+        private void Load() {
+            var trackStream = File.OpenRead(AudioPath);
+            var track = new TrackBass(trackStream);
+            Audio.AddItem(track);
+            var drawableTrack = new DrawableTrack(track);
+            StoryClock = new StopwatchClock();
+            FramedClock = new FramedClock(StoryClock);
```
Comment (maxrchung) [inline S2VX.Game.Tests/VisualTests/EditorHoldNoteTests.cs:37]: Don't need `FramedClock` property if it's not used.

File: S2VX.Game.Tests/VisualTests/EditorHoldNoteTests.cs [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+            Story.Clock = FramedClock;
+            Editor = new EditorScreen(null, Story, drawableTrack);
+            ScreenStack.Push(Editor);
```
Comment (maxrchung) [inline S2VX.Game.Tests/VisualTests/EditorHoldNoteTests.cs:40]: Don't need to save `ScreenStack` either

File: S2VX.Game.Tests/VisualTests/EditorHoldNoteTests.cs [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+            Add(ScreenStack);
+        }
+
+        // All tests will have a hold note that starts to appear in 1 second and lasts for 1 second
+        [SetUp]
+        public void Setup() {
+            Story.ApplyDefaultCommands(); // Ensure default commands are applied before using any properties
```
Comment (maxrchung) [inline S2VX.Game.Tests/VisualTests/EditorHoldNoteTests.cs:47]: Why is this needed? Story's `Update()` should automatically be called by osu!framework to do this.

File: S2VX.Game.Tests/VisualTests/EditorHoldNoteTests.cs [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+            FramedClock = new FramedClock(StoryClock);
+            Story.Clock = FramedClock;
+            Editor = new EditorScreen(null, Story, drawableTrack);
+            ScreenStack.Push(Editor);
+            Add(ScreenStack);
+        }
+
+        // All tests will have a hold note that starts to appear in 1 second and lasts for 1 second
+        [SetUp]
+        public void Setup() {
```
Comment (maxrchung) [inline S2VX.Game.Tests/VisualTests/EditorHoldNoteTests.cs:46]: If you set up your own `[SetUp]` function, I think this ignores the default `SetUp()` function inside of `S2VXTestScene` and its parent classes.

To fully override `SetUp()`, you should rename this `public new void SetUp()` and call `base.SetUp()` in the function.

File: S2VX.Game.Tests/VisualTests/EditorHoldNoteTests.cs [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+        [Test]
+        public void EditorHoldNoteAlpha_BeforeFadeInTime_IsZero() {
+            AddStep("Seek before FadeInTime", () => Editor.Seek(NoteAppearTime));
+            AddAssert("Note is not visible", () => NoteToTest.Alpha == 0);
+        }
+
+        [Test]
+        public void EditorHoldNoteAlpha_AfterFadeInBeforeShowTime_IsBetweenZeroAndOne() {
+            AddStep("Seek between FadeInTime and ShowTime", () => Editor.Seek(NoteAppearTime + Story.Notes.FadeInTime / 2));
+            AddAssert("Note is partially visible", () => NoteToTest.Alpha is > 0 and < 1);
```
[the remark this answers is not in the document]
Comment (Xenocidel) [reply S2VX.Game.Tests/VisualTests/EditorHoldNoteTests.cs:64]: Yeah this was auto-corrected based on this rule
https://docs.microsoft.com/en-us/dotnet/fundamentals/code-analysis/style-rules/ide0078

File: S2VX.Game.Tests/S2VX.Game.Tests.csproj [hunk 6]
```diff
@@ -32,6 +32,12 @@
     <PackageReference Include="ppy.osu.Framework" Version="2021.215.0" />
   </ItemGroup>
   <ItemGroup>
+    <None Update="VisualTests\EditorTests\10-seconds-of-silence.mp3">
+      <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
+    </None>
+    <None Update="VisualTests\EditorTests\EditorHoldNote.s2ry">
```
[the remark this answers is not in the document]
Comment (Xenocidel) [reply S2VX.Game.Tests/S2VX.Game.Tests.csproj:38]: Removed in next commit

File: S2VX.Game.Tests/VisualTests/EditorHoldNoteTests.cs (hunk 1 above)
Comment (Xenocidel) [reply S2VX.Game.Tests/VisualTests/EditorHoldNoteTests.cs:37]: Removed in next commit

File: S2VX.Game.Tests/VisualTests/EditorHoldNoteTests.cs (hunk 2 above)
Comment (Xenocidel) [reply S2VX.Game.Tests/VisualTests/EditorHoldNoteTests.cs:40]: Removed in next commit

File: S2VX.Game.Tests/VisualTests/EditorHoldNoteTests.cs (hunk 3 above)
Comment (Xenocidel) [reply S2VX.Game.Tests/VisualTests/EditorHoldNoteTests.cs:47]: There is a race condition between `Update()` and when the test accesses the properties otherwise. If you debug the test and look at Story.Notes.FadeInTime for example, it will be 0. The whole reason why I needed to bring this out into a public function was so that I ensure the defaults are applied before I use them on the very next line.

File: S2VX.Game.Tests/VisualTests/EditorHoldNoteTests.cs (hunk 4 above)
Comment (Xenocidel) [reply S2VX.Game.Tests/VisualTests/EditorHoldNoteTests.cs:46]: Done in next commit

File: S2VX.Game/Story/S2VXStory.cs [hunk 7]
```diff
@@ -165,7 +165,7 @@ public class S2VXStory : CompositeDrawable {
             File.WriteAllText(path, serialized);
         }
 
-        protected override void Update() {
+        public void ApplyDefaultCommands() {
```
[the remark this answers is not in the document]
Comment (Xenocidel) [reply S2VX.Game/Story/S2VXStory.cs:168]: Adjusted this function to only include the default command adding part

File: S2VX.Game.Tests/VisualTests/EditorHoldNoteTests.cs (hunk 5 above)
[the remark this answers is not in the document]
Comment (maxrchung) [reply S2VX.Game.Tests/VisualTests/EditorHoldNoteTests.cs:65]: Pretty amazing

File: S2VX.Game/Story/S2VXStory.cs (hunk 7 above)
[the remark this answers is not in the document]
Comment (maxrchung) [reply S2VX.Game/Story/S2VXStory.cs]: I refactored/removed this in favor of using `ScheduleAfterChildren()`

File: S2VX.Game.Tests/VisualTests/EditorHoldNoteTests.cs (hunk 3 above)
Comment (maxrchung) [reply S2VX.Game.Tests/VisualTests/EditorHoldNoteTests.cs]: Refactored this so that we are using `ScheduleAfterChildren()`

**Review by maxrchung — APPROVED** [review]

**Review by Naranja-Sagged — APPROVED** [review]
